### PR TITLE
Fix: allocation carry over

### PR DESCRIPTION
**Pull request by shark0der**
## Context

Issue #872 

## Changes proposed in this pull request

Carry over allocations from previous tranches.

## Test plan

Added a test that fails with the current release-candidate and passes with the fix.

## Checklist

- [ ] Rebased the base branch
- [ ] Attached corresponding Github issue
- [ ] Prefixed the name with the type of change (i.e. feat, chore, test)
- [ ] Performed a self-review of my own code
- [ ] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [ ] Didn't generate new warnings
- [ ] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: contracts/modules/staking/StakingPool.sol [hunk 1]
```diff
@@ -922,31 +922,61 @@ contract StakingPool is IStakingPool, Multicall {
 
       uint[] memory trancheCapacities = getTrancheCapacities(
         request.productId,
-        firstTrancheIdToUse,
-        MAX_ACTIVE_TRANCHES - startIndex, // count
+        _firstActiveTrancheId,
```
Comment (danoctavian) [inline contracts/modules/staking/StakingPool.sol:925]: what's the logic behind starting with _firstActiveTrancheId vs firstTrancheIdToUse

Comment (shark0der) [reply contracts/modules/staking/StakingPool.sol:925]: - first active tranche id - exactly what it says, from all active tranches it has the lowest id
- first active to use - from all active ones, the first one that expires after the cover grace period, basically the first one we can use for the current cover

File: test/unit/StakingPool/requestAllocation.js [hunk 2]
```diff
@@ -1530,4 +1530,117 @@ describe('requestAllocation', function () {
...
+        secondCoverAmount.add(1), // slightly over the limit
+        previousPremium,
+        unfullfillableRequest,
+      ),
+    ).to.be.revertedWithCustomError(stakingPool, 'InsufficientCapacity');
+
+    stakingPool.connect(this.coverSigner).requestAllocation(
+      secondCoverAmount, // exact available amount
+      previousPremium,
+      unfullfillableRequest,
```
Comment (danoctavian) [inline test/unit/StakingPool/requestAllocation.js:1643]: this one is actually fulfillable right?

it doesn't revert

**Review by roxdanila — APPROVED** [review]

File: test/unit/StakingPool/requestAllocation.js (hunk 2 above)
Comment (shark0der) [reply test/unit/StakingPool/requestAllocation.js]: Adjusted the variable names.

**Pull request merged**